linera-views/src/batch.rs

```diff
@@ -158,29 +159,39 @@ impl Batch {
         for op in self.operations {
             match op {
                 WriteOperation::Delete { key } => {
+                    // We delete a key. However if said key was already matched by a delete_prefix then
+                    // nothing needs to be done.
                     if !is_prefix_matched(&delete_prefix_set, &key) {
                         delete_and_insert_map.insert(key, None);
                     }
                 }
                 WriteOperation::Put { key, value } => {
+                    // Simple insert
                     delete_and_insert_map.insert(key, Some(value));
                 }
                 WriteOperation::DeletePrefix { key_prefix } => {
```

Comment:
Looks like what follows is one of the bug fixes. Can we add a unit test on `Batch::simplify`? (And by the book it should cover the bug.)

Reply:
Ok, added.

Reply:
Ok, done.